### DISCUSSION TITLE
Support for services with ROS 1 native connection

### DISCUSSION THREAD

**Discussion by meropis** [foxglove/community#446] [Q&A]
**Description**
I am attempting to call a service in ROS1 using `context.callService`, but my linter throws the above error. To fix this I used optional chaining e.g. `context.callService?.()` which no longer threw the error at me, but when I run the panel it doesn't call the service at all. Do I have to do something else before I use this method?

- Version: latest
- OS: ubuntu 22.04 LTS
- Data source: ROS1

**Steps To Reproduce**
Create a basic panel, import `Button` from `@mui/material`. Within the button click event, do `context.callService("example_service", { foo: "bar" })` - it will error (granted you are using typescript and a linter).
```js
import { Button } from "@mui/material";
```
Button design
```html
<Button
        variant="contained"
        color="success"
        title="call the service"
        onClick={() =>
          context.callService("example_service", { foo: "bar" }) //typescript shouts the error at me here
        }
      >
        call service
</Button>
```

**Expected Behavior**
calls a service (that definitely exists)

**Actual Behavior**
no service call, no error

**Comment by jtbandes** [comment]
The ROS 1 native connection does not currently support services. Try using Rosbridge instead. See compatibility table here: https://foxglove.dev/docs/studio/connection/data-sources
